log start time and config

Gueorgi opened the pull request:
# Description

Log info about the start-up time and configuration used when connections are being opened. The existing logging of such info in RxDocumentClientImpl can't calculate the total time since it is missing few calls before and after in the flow and was not printing all the info.

The logged info looks like this (sample taken from local run with this change):

`INFO com.azure.cosmos.CosmosClientBuilder - Cosmos Client with (Correlation) ID [00005] started up in [335] ms with the following configuration: serviceEndpoint [https://forklift1-dev-azure-cosmosdb.documents.azure.com:443/], preferredRegions [[]], connectionPolicy [ConnectionPolicy{httpNetworkRequestTimeout=PT1M, tcpNetworkRequestTimeout=PT5S, connectionMode=DIRECT, maxConnectionPoolSize=1000, idleHttpConnectionTimeout=PT1M, idleTcpConnectionTimeout=PT0S, userAgentSuffix='-wmt-azsdk-java-cosmos/4.41.0-beta.1 MacOSX/12.6.2 JRE/1.8.0_333', throttlingRetryOptions=RetryOptions{maxRetryAttemptsOnThrottledRequests=9, maxRetryWaitTime=PT30S}, endpointDiscoveryEnabled=false, preferredRegions=null, multipleWriteRegionsEnabled=false, proxyType=null, inetSocketProxyAddress=null, readRequestsFallbackEnabled=false, connectTimeout=PT5S, idleTcpEndpointTimeout=PT1H, maxConnectionsPerEndpoint=130, maxRequestsPerConnection=30, tcpConnectionEndpointRediscoveryEnabled=true}], consistencyLevel [Session], contentResponseOnWriteEnabled [true], sessionCapturingOverride [false], connectionSharingAcrossClients [false], clientTelemetryEnabled [false].`

There are no tests since looks like logging info is not verified anywhere in the SDK (if there are samples let me know) and the above snapshot was the result of the change being running.

Commit:
log start time and config

Gueorgi commented on sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java:
```diff
@@ -374,10 +374,12 @@ private RxDocumentClientImpl(URI serviceEndpoint,
-        logger.info(
-            "Initializing DocumentClient [{}] with"
-                + " serviceEndpoint [{}], connectionPolicy [{}], consistencyLevel [{}], directModeProtocol [{}]",
-            this.clientId, serviceEndpoint, connectionPolicy, consistencyLevel, configs.getProtocol());
```
Avoid potentially costly conversion of object to a string for logging if the info is not going to be
logged anyway. One should not call logger.info/debug/warn unless the right log leve is enabled. The rest of
the logger.info/debug/warn calls in this class are not taking much CPU even if wasted so are left as is.

JJ-WM replied:
is check info necessary since it's the default setting in most cases?

Gueorgi replied:
Yes, some apps run in warn level and up only - no info. So why incur the cost if it is not going to be logged?

Gueorgi commented on sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java:
```diff
@@ -21,6 +24,10 @@
+// TODO: Use slf4j Logger and LoggerFactory instead of the core ClientLogger?
+// import org.slf4j.Logger;
+// import org.slf4j.LoggerFactory;
+
```
ClientLogger (a core API class) is being used in the repo ... but so is log4j factory/logger.
I am not sure what would be the better option. Seems one should be using ClientLogger.
I owul remove this commented out import or uncomment it and remove the ClientLogger once it is clear what to use.

Gueorgi replied:
Switched to log factory like used in the rest of cosmos SDK

Gueorgi committed:
Update CosmosClientBuilder.java

Remove user suffix since is being printed out as part of the connection policy.

Gueorgi committed:
Merge branch 'Azure:main' into log-startup-time-and-config

mikeharder committed:
[Perf] Upgrade perf pipeline to Java 17 (#31407)

Commit:
remove rx clinet info log

Gueorgi committed:
Update PhoneNumbersClientBuilder.java

Gueorgi committed:
Update PhoneNumbersClient.java

Gueorgi committed:
Update perf.yml

Gueorgi committed:
Update perf.yml

Commit:
spacing in comments

Commit:
Merge remote-tracking branch 'origin/main' into log-startup-time-and-config

Commit:
Merge remote-tracking branch 'origin/main' into log-startup-time-and-config

Commit:
sync main and use log factory

Gueorgi commented:
Code was merged in main branch of the azure SDK git repo
